### PR TITLE
 feat(`cheatcodes`): add `delegatecall` to `prank`ing

### PR DESCRIPTION
**Pull request by EdwardJES**
## Motivation
Closes https://github.com/foundry-rs/foundry/issues/824. Currently there is no easy way to test delegate calls using foundry tests.

## Solution
- Add a `delegate_call` flag to the prank.
- ....

### CODE REVIEW

File: crates/cheatcodes/src/inspector.rs [hunk 1]
```diff
@@ -833,6 +833,19 @@ impl Cheatcodes {
 
         // Apply our prank
         if let Some(prank) = &self.prank {
+            // Apply delegate call. call.caller will not equal prank.prank_caller
```
Comment (EdwardJES) [inline crates/cheatcodes/src/inspector.rs:836]: Hey @zerosnacks, my understanding is a little fuzzy here. It seems in the case of a delegate call that `call.caller == prank.prank_caller` will not be true. Is it suitable to have this here? Ty.

Comment (zerosnacks) [reply crates/cheatcodes/src/inspector.rs:836]: cc @DaniPopes would you have more context on this?

**Comment by zerosnacks** [comment]
Hi @EdwardJES thanks for your PR - my apologies that it has taken a while to respond

**Comment by EdwardJES** [comment]
gentle bump on this @zerosnacks, @DaniPopes . Thank you 🙏

**Comment by grandizzy** [comment]
@EdwardJES is there anything waiting from us to unblock the PR or can we start the review? thank you!

**Comment by EdwardJES** [comment]
@grandizzy all good to go, thanks!

File: crates/cheatcodes/src/inspector.rs [hunk 2]
```diff
@@ -941,6 +941,19 @@ where {
...
+            if let CallScheme::DelegateCall = call.scheme {
```
Comment (grandizzy) [inline crates/cheatcodes/src/inspector.rs:945]: should we handle `CallScheme::ExtDelegateCall` here too?

**Review by grandizzy — COMMENTED** [review]
looks good, thank you! I rebased and added a negative test (EOA prank), have one comment re handling `ExtDelegateCall` as well

File: crates/cheatcodes/src/inspector.rs (hunk 2 above)
Comment (EdwardJES) [reply crates/cheatcodes/src/inspector.rs]: Good catch, yes.

**Review by grandizzy — APPROVED** [review]
lgtm, thank you! pending other reviews before merging

**Review by mattsse — APPROVED** [review]
lgtm

**Comment by leovct** [comment]
I tried to use this new feature but it doesn't work since I guess it requires an update of [forge-std](https://github.com/foundry-rs/forge-std)?

```bash
Error (9582): Member "startPrank" not found or not visible after argument-dependent lookup in contract Vm.
  --> test/EthernautCTF/DelegationExploit.t.sol:26:5:
   |
26 |     vm.startPrank(exploiter, true);
   |     ^^^^^^^^^^^^^

Error: Compilation failed
```

**Comment by grandizzy** [comment]
indeed, will update and release a version with, thanks for flagging

**Comment by varun-doshi** [comment]
Hi
Wondering if delegatePrank is now available? I couldn't find it in the book

**Comment by grandizzy** [comment]
@varun-doshi yes it is, book is not updated yet

**Pull request merged**